src/api/MainSolver.h

```diff
@@ -68,136 +51,162 @@ const sstat s_Error = toSstat( 2);
 
 class MainSolver
 {
-  protected:
+protected: /** Helper classes to deal with assertion stack, preprocessing and substitutions **/
+    struct FrameId
```

Comment:
(And it will somewhat unite it withCnfizer)

Reply:
Sometimes, it is good to have strong types. It can make the interfaces cleaner and safer.
But it looks like we do not really need it, so maybe we can remove the struct here.

Reply:
Ok, sounds good!